src/App.js

```diff
@@ -198,51 +235,54 @@ class FeatureLoader extends React.Component<Props, State> {
     }
   }
 
-  async componentWillMount() {
+
+  async componentDidMount(): Promise<void> {
     this.onHashUpdate();
     window.addEventListener('resize', this.resizeListener);
     this.resizeListener();
     await loadExistingLocalizationByPreference()
       .then(() => this.setState({ isLocalizationLoaded: true }))
-  }
-
-
-  componentDidMount() {
     window.addEventListener('hashchange', this.onHashUpdate);
-    this.updateStateFromProps(this.props);
+    this.fetchFeature(getFeatureIdFromProps(this.props));
   }
 
 
   componentDidUpdate(prevProps, prevState) {
     this.manageFocus(prevProps, prevState);
+    if (featureIdHasChanged(this.props, prevState)) {
+      this.fetchFeature(getFeatureIdFromProps(this.props));
+    }
   }
 
 
-  updateStateFromProps(props: Props) {
-    this.fetchFeature(props);
+  static getDerivedStateFromProps(newProps: Props, prevState: State): State {
+    const result: $Shape<State> = {};
 
-    if (this.featureId(props) !== this.featureId(this.props)) {
-      this.setState({ isFilterToolbarVisible: false });
+    if (featureIdHasChanged(newProps, prevState)) {
+      result.isFilterToolbarVisible = false;
+      result.featureId = getFeatureIdFromProps(newProps);
```

Comment:
-1: This will always be a string or null, remove `number` from type in state

Reply:
Done!